vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -23,8 +23,12 @@ def save_inventory_multi(type, parent, hashes, deletes, find_key, child_keys = [
 
   def save_inventory_single(type, parent, hash, child_keys = [], extra_keys = [])
     child_keys, extra_keys, remove_keys = self.save_inventory_prep(child_keys, extra_keys)
-    save_inventory(type, parent, hash.except(*remove_keys))
-    save_child_inventory(parent.send(type), hash, child_keys)
+    if hash.nil?
```

Comment:
Let's change this to `if hash.blank?`  This would handle the `:hardware => {}` case in https://github.com/ManageIQ/manageiq/pull/1376#issuecomment-77620135, which although questionable in usage, would at least not lead to head scratching if it were to occur.